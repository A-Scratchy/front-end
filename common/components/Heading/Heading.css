.Heading {
  display: flex;
  flex-direction: row;
  justify-content: center;
  text-align: center;
  width: 85%;
}

.headingLines:before,
.headingLines:after {
  flex-grow: 1;
  height: 3px;
  content: "\a0";
  background-color: var(--primary);
  align-self: center;
  margin: 0 10px 5px 10px;
}

.dark {
  color: var(--slate);
}

.white {
  color: #ffffff;
}
