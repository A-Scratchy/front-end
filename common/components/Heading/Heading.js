import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import styles from './Heading.css';

Heading.propTypes = {
  id: PropTypes.string,
  text: PropTypes.string.isRequired,
  headingLines: PropTypes.bool,
  theme: PropTypes.string,
};

Heading.defaultProps = {
  id: null,
  headingLines: true,
  theme: 'dark',
};

function Heading({
  text, headingLines, theme, ...otherProps
}) {
  const classes = classNames({
    [`${styles.Heading}`]: true,
    [`${styles[theme]}`]: true,
    [`${styles.headingLines}`]: headingLines,
  });

  return (
    <h2
      className={classes}
      {...otherProps}
    >
      {text}
    </h2>
  );
}

export default Heading;
