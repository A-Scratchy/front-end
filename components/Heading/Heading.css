.Heading {
  text-transform: uppercase;
  margin: 1rem 0;
}

.headingContainer {
  display: flex;
  justify-content: center;
}

.headingTextWithLinkIconOffset {
  margin-right: 26px;
}

.headingOurMission {
  display: flex;
  flex-direction: row;
  justify-content: center;
  text-align: center;
  text-transform: uppercase;
  width: 85%;
  margin: 1rem auto;
}

.headingLines::before,
.headingLines::after {
  flex-grow: 1;
  height: 3px;
  content: "\a0";
  background-color: var(--primary);
  align-self: center;
  margin: 0 10px 5px 10px;
}

.secondaryFont {
  color: var(--secondary);
}

.grayFont {
  color: var(--gray);
}

.whiteFont {
  color: var(--white);
}

.icon {
  width: 18px;
  pointer-events: none;
}

:target {
  content: "";
  display: block;
  height: 100px;
  margin-top: -100px;
}

.anchor {
  line-height: 2.25;
  height: 2rem;
  margin: 1rem 0.25rem;
  outline: none;
}

.iconFillWhite {
  fill: #f7f7f7;
}

.iconFillBlue {
  fill: #252e3e;
}

.iconVisible {
  visibility: visible;
}

.iconHidden {
  visibility: hidden;
}

@media screen and (--medium-viewport) {
  .Heading {
    width: 90%;
  }
}

@media screen and (--small-viewport) {
  .Heading {
    width: 100%;
  }
}
