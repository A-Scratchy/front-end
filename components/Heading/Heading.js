import React, { Component } from 'react';
import { string, number, bool } from 'prop-types';
import classNames from 'classnames';
import kebabCase from 'lodash/kebabCase';
import ScreenReaderOnly from 'components/ScreenReaderOnly/ScreenReaderOnly';
import LinkIcon from 'static/images/icons/FontAwesome/link-solid.svg';
import styles from './Heading.css';

class Heading extends Component {
  static propTypes = {
    className: string,
    customAnchorClass: string,
    text: string.isRequired,
    hasTitleUnderline: bool,
    hasHeadingLines: bool,
    hasHashLink: bool,
    headingLevel: number,
  };

  static defaultProps = {
    className: undefined,
    customAnchorClass: 'anchorMargin',
    hasTitleUnderline: false,
    hasHeadingLines: false,
    hasHashLink: true,
    headingLevel: 2,
  };

  render() {
    const { props } = this;
    const anchorId = kebabCase(props.text);
    const HeadingElement = `h${props.headingLevel}`;

    return (
      <div className={`${styles.headingContainer}`}>
        {props.hasHashLink && (
          <>
            <span
              id={anchorId}
              className={classNames(styles.anchorSpan, styles[props.customAnchorClass])}
            />
            <div className={`${styles.hashLinkContainer}`}>
              <HeadingElement
                className={classNames(
                  props.className,
                  styles.headingTextWithLinkIconOffset,
                  styles.Heading,
                )}
              >
                <div className={`${styles.hashLink}`}>
                  <a id={`${anchorId}-link`} href={`#${anchorId}`} data-testid="Hash Link">
                    <ScreenReaderOnly>Scroll Link for {props.text}</ScreenReaderOnly>
                    <LinkIcon className={styles.icon} />
                  </a>
                  {props.hasTitleUnderline ? (
                    <span className={`${styles.underline}`}>{props.text}</span>
                  ) : (
                    <span>{props.text}</span>
                  )}
                </div>
              </HeadingElement>
            </div>
          </>
        )}
        {props.hasHeadingLines && (
          <HeadingElement className={classNames(styles[props.className], styles.headingLines)}>
            {props.text}
          </HeadingElement>
        )}
      </div>
    );
  }
}

export default Heading;
