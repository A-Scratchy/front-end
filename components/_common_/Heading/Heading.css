@import "../../../common/styles/media-queries.css";

.Heading {
  display: flex;
  flex-direction: row;
  justify-content: center;
  margin-bottom: 1rem;
  text-align: center;
  text-transform: uppercase;
  width: 85%;
}

.headingLines::before,
.headingLines::after {
  flex-grow: 1;
  height: 3px;
  content: "\a0";
  background-color: var(--primary);
  align-self: center;
  margin: 0 10px 5px 10px;
}

.secondary {
  color: var(--secondary);
}

.gray {
  color: var(--gray);
}

.white {
  color: var(--white);
}

@media screen and (--medium-viewport) {
  .Heading {
    width: 90%;
  }
}

@media screen and (--small-viewport) {
  .Heading {
    width: 100%;
  }
}
