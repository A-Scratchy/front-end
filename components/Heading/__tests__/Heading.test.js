import React from 'react';
import { shallow } from 'enzyme';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Heading from '../Heading';

describe('Heading', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Heading>Test</Heading>);
  });

  it('should render with many props assigned', () => {
    createSnapshotTest(
      <Heading className="test-class" id="test-heading-1" hasHeadingLines={false} theme="secondary">
        Test
      </Heading>,
    );
  });

  it('should render with "secondary" in classNames when theme="secondary"', () => {
    const HeaderInstance = shallow(<Heading theme="secondary">Test</Heading>);

    expect(HeaderInstance).toHaveClassName('secondary');
  });

  it('should render without "headingLines" in classNames when hasHeadingLines={false}', () => {
    const HeaderInstance = shallow(<Heading hasHeadingLines={false}>Test</Heading>);

    expect(HeaderInstance).not.toHaveClassName('headingLines');
  });
});



import React from 'react';
// import { shallow } from 'enzyme';
// import createSnapshotTest from 'test-utils/createSnapshotTest';

// import HashLink from '../HashLink';

// describe('HashLink', () => {
//   it('should render with required props', () => {
//     createSnapshotTest(<HashLink>Test</HashLink>);
//   });

//   it('should render with many props assigned', () => {
//     createSnapshotTest(<HashLink className="test-class">Test</HashLink>);
//   });

//   it('should contain filtered anchorId', () => {
//     const questionMarkProperties = {
//       id: 'WANT TO BECOME A SPONSOR?',
//     };

//     const exclamationMarkProperties = {
//       id: 'JOIN TODAY!',
//     };

//     const questionMarkWrapper = shallow(<HashLink {...questionMarkProperties} />);
//     const exclamationMarkWrapper = shallow(<HashLink {...exclamationMarkProperties} />);

//     expect(questionMarkWrapper.instance().getAnchorId()).toStrictEqual('want-to-become-a-sponsor');
//     expect(exclamationMarkWrapper.instance().getAnchorId()).toStrictEqual('join-today');
//   });

//   it('should contain the anchor classes', () => {
//     const defaultProps = {
//       customIconOffset: 'default',
//     };

//     const offsetProperties = {
//       customIconOffset: 'offsetLineHeightOne',
//     };

//     const defaultWrapper = shallow(<HashLink {...defaultProps} />);
//     const offsetWrapper = shallow(<HashLink {...offsetProperties} />);

//     expect(defaultWrapper.instance().getAnchorClass()).toStrictEqual('anchorDefault');
//     expect(offsetWrapper.instance().getAnchorClass()).toStrictEqual('anchorOffsetLineHeightOne');
//   });

//   it('should contain visible icon class', () => {
//     const whiteProperties = {
//       theme: 'white',
//     };

//     const blueProperties = {
//       theme: 'blue',
//     };

//     const whiteWrapper = shallow(<HashLink {...whiteProperties} />);
//     const blueWrapper = shallow(<HashLink {...blueProperties} />);

//     expect(whiteWrapper.instance().getVisibleIcon()).toStrictEqual('icon iconVisibleiconFillBlue');
//     expect(blueWrapper.instance().getVisibleIcon()).toStrictEqual('icon iconVisibleiconFillWhite');
//   });

//   it('should toggle the visible link icon', () => {
//     const wrapper = shallow(<HashLink />);
//     const instance = wrapper.instance();

//     instance.toggleVisible(true);
//     expect(instance.state.isVisible).toStrictEqual(true);

//     instance.toggleVisible(false);
//     expect(instance.state.isVisible).toStrictEqual(false);
//   });

//   it('should contain id link', () => {
//     const questionMarkProperties = {
//       id: 'WANT TO BECOME A SPONSOR?',
//     };

//     const exclamationMarkProperties = {
//       id: 'JOIN TODAY!',
//     };

//     const questionMarkWrapper = shallow(<HashLink {...questionMarkProperties} />);
//     const exclamationMarkWrapper = shallow(<HashLink {...exclamationMarkProperties} />);

//     expect(questionMarkWrapper.find('#want-to-become-a-sponsor')).toExist(true);
//     expect(exclamationMarkWrapper.find('#join-today')).toExist(true);
//   });

//   it('should contain hashlink in href attribute', () => {
//     const requiredProps = {
//       id: 'GENERAL QUESTIONS',
//     };

//     const wrapper = shallow(<HashLink {...requiredProps} />);

//     expect(wrapper.find({ href: '#general-questions' })).toExist(true);
//   });
// });

