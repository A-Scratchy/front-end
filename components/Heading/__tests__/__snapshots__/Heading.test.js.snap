// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should render with many props assigned 1`] = `
<div
  className="headingContainer"
>
  <span
    className="anchorSpan anchorMargin"
    id="test"
  />
  <div
    className="hashLinkContainer"
  >
    <h2
      className="secondary test-class headingTextWithLinkIconOffset Heading"
    >
      <div
        className="hashLink"
      >
        <a
          data-testid="Hash Link"
          href="#test"
          id="test-link"
        >
          <span
            className="ScreenReaderOnly"
          >
            Scroll Link for 
            Test
          </span>
          <svg
            className="icon"
          />
        </a>
        <span>
          Test
        </span>
      </div>
    </h2>
  </div>
</div>
`;

exports[`Heading should render with required props 1`] = `
<div
  className="headingContainer"
>
  <span
    className="anchorSpan anchorMargin"
    id="test"
  />
  <div
    className="hashLinkContainer"
  >
    <h2
      className="secondary headingTextWithLinkIconOffset Heading"
    >
      <div
        className="hashLink"
      >
        <a
          data-testid="Hash Link"
          href="#test"
          id="test-link"
        >
          <span
            className="ScreenReaderOnly"
          >
            Scroll Link for 
            Test
          </span>
          <svg
            className="icon"
          />
        </a>
        <span>
          Test
        </span>
      </div>
    </h2>
  </div>
</div>
`;
